src/index.ts

```diff
@@ -3,6 +3,37 @@ import { verifyHash } from "./hash/hash";
 import { verifyIssued } from "./issued/verify";
 import { verifyRevoked } from "./revoked/verify";
 import { documentToSmartContracts } from "./common/smartContract/documentToSmartContracts";
+import { OpenAttestationContract } from "./types";
+
+type ResolveType<T> = T extends (...args: any[]) => Promise<infer U> ? U : T;
```

Comment:
I think this one deserve a comment to explain what we are trying (comment + example maybe)

Reply:
OK, I've added some comments about this